### ISSUE TITLE
[Bug]: Libjuju errors out if simple entries are present in a charmhub charm's metadata

### ISSUE THREAD

**Issue by dragomirp** [juju/python-libjuju#850]
### Description

In my case the error seems to be caused if assumes juju or assumes k8s-api are present in the charm's metadata

I think the root cause is the changes made to facade.py in https://github.com/juju/python-libjuju/commit/4eb26c7dd46508dcbc7d85950407ebcd465dccbf

Errors in Github CI:
- [Assumes k8s-api](https://github.com/canonical/pgbouncer-k8s-operator/actions/runs/4954060987/jobs/8909145210)
- [Assumes juju](https://github.com/canonical/pgbouncer-operator/actions/runs/4979717669/jobs/8911712013)

Traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "<console>", line 1, in <module>
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/model.py", line 1664, in deploy
    resources = await self._add_charmhub_resources(res.app_name,
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/model.py", line 1797, in _add_charmhub_resources
    res = await charm_facade.CharmInfo(entity_url)
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/facade.py", line 502, in wrapper
    result = cls.from_json(reply['response'])
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/facade.py", line 673, in from_json
    return cls(**d)
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/_definitions.py", line 4890, in __init__
    meta_ = CharmMeta.from_json(meta) if meta else None
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/facade.py", line 673, in from_json
    return cls(**d)
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/_definitions.py", line 5402, in __init__
    assumes_expr_ = ExpressionTree.from_json(assumes_expr) if assumes_expr else None
  File "/home/ubuntu/workspace/pgbouncer-operator/.tox/backend-integration/lib/python3.10/site-packages/juju/client/facade.py", line 692, in from_json
    cls.splitEntries(v, d)
UnboundLocalError: local variable 'v' referenced before assignment
```

### Urgency

Annoying bug in our test suite

### Python-libjuju version

2.9.42.2

### Juju version

2.9.42

### Reproduce / Test

```python
import asyncio
from juju.model import Model
model = Model()
await model.connect_current()

await model.deploy("postgresql", channel="14/edge")
```

**Comment by taurus-forever** [comment]
Hi @cderici , @juanmanuel-tirado is not available this week, he said you can check the issue
(at the moment Renovate created several dependency update PRs in our repos, all blocked due to this bug).

No rush necessary, just a question about ETA to make a decision on our PRs (wait for the fix, or extract libjuju from PRs). Tnx!

**Comment by cderici** [comment]
Hi @taurus-forever & @dragomirp, thanks for reporting this, just pushed a fix that should solve this. Feel free to reach out if it doesn't so we'll try to unblock you asap 👍

**Comment by cderici** [comment]
fixed by #854
